src/IonDecimal.ts

```diff
@@ -49,27 +49,68 @@ export class Decimal {
     public static readonly ONE = new Decimal(1, 0);
 
     /**
-     * Creates a new Decimal value.
-     * @param coefficient   See the {@link Decimal} class-level documentation for details.
-     * @param exponent      See the {@link Decimal} class-level documentation for details.
+     * Creates a new Decimal value. Three parameter sets are supported:
+     *
+     *  (text: string)
+     *      The provided text will be parsed into a Decimal value.
+     *
+     *  (coefficient: number, coefficient: number)
+     *      A Decimal value will be constructed using the provided coefficient and exponent values.
+     *
+     *  (coefficient: BigInt, coefficient: number, isNegative?: boolean)
+     *      A Decimal value will be constructed using the provided coefficient and exponent values.
+     *      Because the BigInt data type cannot represent -0, a third parameter called `isNegative` may be supplied
+     *      to explicitly set the sign of the Decimal following construction. If this flag is not specified,
+     *      the provided coefficient's sign will be used.
+     */
+    constructor(coefficient: number | JSBI | string,
+                exponent?: number,
+                isNegative?: boolean) {
+        if(typeof coefficient === "string") {
+            return Decimal.parse(coefficient);
+        }
+
+        if (!_hasValue(exponent)) {
+            throw new Error("Decimal's constructor was called with a numeric coefficient but no exponent.");
+        }
+
+        if (typeof coefficient === "number") {
+            return Decimal._fromNumberCoefficient(coefficient, exponent);
+        }
```

Comment:
Oh interesting, what is supposed to happen if `number` is a coefficient and `isNegative` is set?

Reply:
I just learned that Typescript allows you to declare which subsets of a signature's possible parameter combinations are permitted with the following syntax:
```typescript
    // Publicly visible/usable signatures
    constructor(coefficient: string);
    constructor(coefficient: number, exponent: number);
    constructor(coefficient: JSBI, exponent: number, isNegative?: boolean);

    // Consolidated, non-public implementation signature
    constructor(coefficient: number | JSBI | string, exponent?: number, isNegative?: boolean) {
```

I'm going to take advantage of this feature to eliminate the usage of unintended combinations.